src/components/scene/device-motion-permission-ui.js

```diff
@@ -0,0 +1,159 @@
+var registerComponent = require('../../core/component').registerComponent;
+var utils = require('../../utils/');
+var bind = utils.bind;
+
+var constants = require('../../constants/');
+
+var DEVICE_PERMISSION_FULL_CLASS = 'a-device-motion-permission-full';
+var DEVICE_PERMISSION_FULL_CENTER_CLASS = 'a-device-motion-permission-full-center';
+var DEVICE_PERMISSION_CONTINUE_CLASS = 'a-device-motion-permission-continue';
+var DEVICE_PERMISSION_CANCEL_CLASS = 'a-device-motion-permission-cancel';
+var BUILT_WITH_AFRAME_CLASS = 'a-built-with-aframe';
+
+/**
+ * UI for enabling device motion permission
+ */
+module.exports.Component = registerComponent('device-motion-permission-ui', {
+  schema: {
+    enabled: { default: true },
+    request: {default: {orientationChange:null, deviceOrientation:null}},
```

Comment:
Does this need to be configurable via schema? Do we need to handle deviceOrientation events explicitly or we can just rely on CSS?

Reply:
No, that was inadvertently left in from something earlier. What is now in place is the single component with the potential for one or both orientation events (device orientation and orientation change). The example on my site has two simple functions attached through the attributes, orientation-change and device-orientation. The drawback that I have on their usage at this point is that they use the window object to reference the functions. I'll hopefully get this done tomorrow with your edits. Thanks for your attention.

<a-scene device-motion-permission-ui orientation-change="noProof" device-orientation="proofOfWork">

https://planetkevin.com/aframe/examples/showcase/device-motion-permission/a.html

Reply:
`
function noProof(event){
--
  | console.log(event);
  | }
  | function proofOfWork(event){
  | console.log(event);
  | }
  | </script>
  |  
  | </head>
  | <body>
  | <a-scene device-motion-permission-ui orientation-change="noProof" device-orientation="proofOfWork">
  |  

`